pkgs/by-name/ot/otb/package.nix

```diff
@@ -0,0 +1,125 @@
+{
+  cmake,
+  callPackage,
+  fetchFromGitHub,
+  makeWrapper,
+  lib,
+  stdenv,
+  swig,
+  which,
+  boost,
+  curl,
+  gdal,
+  libsvm,
+  libgeotiff,
+  muparser,
+  muparserx,
+  opencv,
+  perl,
+  python3,
+  shark,
+  tinyxml,
+  enableFeatureExtraction ? true,
+  enableHyperspectral ? true,
+  enableLearning ? true,
+  enableMiscellaneous ? true,
+  enableOpenMP ? false,
+  enablePython ? true,
+  extraPythonPackages ? ps: with ps; [ ],
+  enableRemote ? true,
+  enableSAR ? true,
+  enableSegmentation ? true,
+  enableStereoProcessing ? true,
```

Comment:
Do we really need those options? or can we just turn them on by default and make things simple?

Reply:
Actually by default OTB has only core modules and all these options are OFF. They have mentioned in doc that incase one needs some module then one can [turn on](https://www.orfeo-toolbox.org/CookBook/CompilingOTBFromSource.html#native-build-with-system-dependencies), but then that requires compiling again from source. I would prefer to keep these option as this is similar to as mentioned in doc, here I kept all `ON` by default (opposite from OTB) as compiling all takes a bit of time (~30min) incase one doesn't need some some module they can override.

Ideally I would prefer something like this in Nixpkgs so that I don't have to build or compile again:
otb (full)
otbMinimal (only core)
otbSAR (otbMinimal + SAR related modules)
otbLearning (otbMinimal + Learning related modules)
...
...